src/ceph-detect-init/ceph_detect_init/__init__.py

```diff
@@ -123,10 +134,23 @@ def platform_information():
         return ('docker', 'docker', 'docker')
 
     if platform.system() == 'Linux':
-        linux_distro = platform.linux_distribution(
-            supported_dists=platform._supported_dists + ('alpine', 'arch'))
-        logging.debug('platform_information: linux_distribution = ' +
-                      str(linux_distro))
+        linux_distro = ('', '', '')
+        if os.path.isfile('/etc/os-release'):
+            try:
+                with open('/etc/os-release', 'r') as f:
+                    data = f.read()
+                linux_distro = (
+                    _extract_from_os_release(data, 'ID'),
+                    _extract_from_os_release(data, 'VERSION'),
+                    '')
+            except Exception as err:
+                logging.debug("platform_information: ",
+                              "Error while opening %s : %s" % (file_name, err))
+        else:
+            linux_distro = platform.linux_distribution(
```

Comment:
have you considered using the `distro` package from pypi?

Reply:
@tchaikov It seems like overkill given that:

* all the Linux distros we're targeting have `/os/release` and are easily testable
* it introduces a new runtime dependency
* ceph-detect-init is deprecated in Mimic and will be dropped in Nautilus